src/DocumentDB.ChangeFeedProcessor/LeaseManagement/PartitionedByLeasePkCollectionRequestOptionsFactory.cs

```diff
@@ -0,0 +1,29 @@
+﻿//----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  Licensed under the MIT license.
+//----------------------------------------------------------------
+
+namespace Microsoft.Azure.Documents.ChangeFeedProcessor.LeaseManagement
+{
+    using System;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor.PartitionManagement;
+    using Microsoft.Azure.Documents.Client;
+
+    /// <summary>
+    /// Used to create request options for partitioned lease collections, when partition key is defined as /leasepk.
+    /// </summary>
+    internal class PartitionedByLeasePkCollectionRequestOptionsFactory : IRequestOptionsFactory
+    {
+        public RequestOptions CreateRequestOptions(ILease lease)
+        {
+            var documentServiceLease = lease as DocumentServiceLease;
+            if (documentServiceLease == null)
+                throw new ArgumentException("lease is not of type DocumentServiceLease");
+            if (string.IsNullOrWhiteSpace(documentServiceLease.LeasePartitionKey))
```

Comment:
This check is technically OK to have as we control setting PK value in the lease document. But in general null/empty PK value is valid.